storage/src/backends/crypto.rs

```diff
@@ -65,10 +66,14 @@ impl<T: Storage> Storage for Backend<T> {
         self.backend.put(hash_key.as_ref().to_vec(), final_value)
     }
 
-    fn delete(&mut self, key: &[u8]) -> Result<()> {
+    fn delete(&self, key: &[u8]) -> Result<()> {
         let hash_key = calculate_sha256(key);
         self.backend.delete(hash_key.as_ref())
     }
+
+    fn prefix_iterator<'a, 'b: 'a>(&'a self, _prefix: &'b [u8]) -> Result<StorageIterator<'a>> {
+        bail!("Iteration is not supported when using encrypted storage")
```

Comment:
This is a problem: it looks like there is a hidden feature that allows to use an encrypted storage for the node. I would try to deprecate that in some way to avoid future problems. Otherwise, we must always assume that iteration can fail.